cpp/src/parquet/column_reader.cc

```diff
@@ -861,8 +864,113 @@ class TypedColumnReaderImpl : public TypedColumnReader<DType>,
   Type::type type() const override { return this->descr_->physical_type(); }
 
   const ColumnDescriptor* descr() const override { return this->descr_; }
+
+  ExposedEncodingType GetExposedEncoding() override { return this->exposed_encoding_; };
+
+  int64_t ReadBatchWithDictionary(int64_t batch_size, int16_t* def_levels,
+                                  int16_t* rep_levels, int32_t* indices,
+                                  int64_t* indices_read, const T** dict,
+                                  int32_t* dict_len) override;
+
+ protected:
+  void SetExposedEncoding(ExposedEncodingType encoding) override {
+    this->exposed_encoding_ = encoding;
+  }
+
+ private:
+  /// \brief Read dictionary indices. Similar to ReadValues but decode data to
```

Comment:
document type comments only belong in header, if this is copy and past you can remove it.

Reply:
you can provide a briefer summary using double slash ("//") comments.

Reply:
thanks, will do